test/e2e/multihoming.go

```
@@ -807,6 +829,113 @@ var _ = Describe("Multi Homing", func() {
+						By("setting new VLAN-ID")
+						netConfig.vlanID = newLocalnetVLANID
+						p := []byte(fmt.Sprintf(`[{"op":"replace","path":"/spec/config","value":%q}]`, generateNADSpec(netConfig)))
+						Expect(patchNADSpec(nadClient, netConfig.name, netConfig.namespace, p)).To(Succeed())
+					})
+
+					It("sets the new MTU on the pod after NetworkAttachmentDefinition reconcile", func() {
+						Eventually(func(g Gomega) error {
+							clientPodConfig := podConfiguration{
+								name:        clientPodName + randStr(10),
+								namespace:   f.Namespace.Name,
+								attachments: []nadapi.NetworkSelectionElement{{Name: secondaryNetworkName}},
+							}
+							kickstartPod(cs, clientPodConfig)
+							mtu, err := getSecondaryInterfaceMTU(clientPodConfig)
+							if err != nil {
+								return fmt.Errorf("failed to get MTU: %w", err)
+							}
+							if mtu != expectedChangedMTU {
+								g.Expect(cs.CoreV1().Pods(clientPodConfig.namespace).Delete(context.Background(), clientPodConfig.name, metav1.DeleteOptions{})).To(Succeed())
+								return fmt.Errorf("pod MTU is %d, but expected %d", mtu, expectedChangedMTU)
+							}
+							return nil
+						}).Should(Succeed(), "pod MTU should be properly configured")
+					})
+
+					It("allocates the pod's secondary interface IP in the new range after NetworkAttachmentDefinition reconcile", func() {
+						By("asserting the pod's secondary interface IP is properly configured")
+						Eventually(func(g Gomega) error {
+							clientPodConfig := podConfiguration{
+								name:        clientPodName + "-" + randStr(10),
+								namespace:   f.Namespace.Name,
+								attachments: []nadapi.NetworkSelectionElement{{Name: secondaryNetworkName}},
+							}
+							kickstartPod(cs, clientPodConfig)
+
+							clientIP, err := podIPForAttachment(cs, clientPodConfig.namespace, clientPodConfig.name, netConfig.name, 0)
+							g.Expect(err).NotTo(HaveOccurred())
```

This is a nit and you may ignore it, but ....

If you are going to be combining returning err with g.Expect inside the Eventually block (which is fine but sets us up a bit for ambiguity, I was originally giving you the two options expecting you to pick one or the other instead of combining them) then I would return error for the main business logic of the Eventually block and use Expect just on side-effects. 

With that I am saying that I would return this error instead of Expecting on it.
Make sense to interpret pod delete as a side effect though.